glusterd2/commands/peers/addpeer.go

```diff
@@ -81,6 +90,14 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	newpeer.MetaData = req.MetaData
+	if req.Zone != "" {
+		if newpeer.MetaData != nil {
```

Comment:
nit: can be simplified

    if newpeer.MetaData == nil {
        newpeer.MetaData = make(map[string]string)
    }
    newpeer.MetaData["_zone"] = req.Zone